### PR TITLE
image-rs: improve AA build and output

### PR DESCRIPTION
**Pull request by katexochen**
Improving output when building and starting AA.
Placing gRPC and ttRPC binaries in different directories, so both can be cached and it is clear what feature the binary supports.

Test itself was already fixed in #271.

### CODE REVIEW

File: image-rs/scripts/build_attestation_agent.sh [hunk 1]
```diff
@@ -9,18 +9,26 @@ set -o errexit
...
+parameters=("KBC=offline_fs_kbc")
 
 [ -n "${BASH_VERSION:-}" ] && set -o errtrace
 [ -n "${DEBUG:-}" ] && set -o xtrace
-[ -n "${TTRPC:-}" ] && parameters+=" ttrpc=true"
+if [[ -n "${TTRPC:-}" ]]; then
+    parameters+=("ttrpc=true")
+    dest_dir_suffix="ttrpc"
+else
+    dest_dir_suffix="grpc"
```
Comment (fitzthum) [inline image-rs/scripts/build_attestation_agent.sh:20]: Will we also need changes to the kata initrd building scripts and peer pods tooling to account for the new location?

File: image-rs/tests/common/mod.rs [hunk 2]
```diff
@@ -59,47 +59,59 @@ pub async fn clean() {
...
                 .kill_on_drop(true)
                 .args(&[
                     "--keyprovider_sock",
                     "unix:///run/confidential-containers/attestation-agent/keyprovider.sock",
                     "--getresource_sock",
                     "unix:///run/confidential-containers/attestation-agent/getresource.sock"
                     ])
-                .spawn()?;
+                .spawn()
+                .expect("Failed to start ttrpc attestation-agent");
```
Comment (fitzthum) [inline image-rs/tests/common/mod.rs:103]: Nothing against this more verbose error but note that currently the `--help` flag will tell you if the AA was built with grpc or ttrpc.

Comment (katexochen) [reply image-rs/tests/common/mod.rs:103]: That can be helpful for local development, but won't help for debugging CI runs.

**Review by arronwy — APPROVED** [review]
Thanks @katexochen  LGTM!

File: image-rs/scripts/build_attestation_agent.sh (hunk 1 above)
Comment (katexochen) [reply image-rs/scripts/build_attestation_agent.sh:20]: I'm not sure I understand that question. I think this build script is only ever called by the image-rs tests. Everything else should use the attestation-agents Makefile. The small modification to the attestation-agent Makefile should only change behavior in edge cases.
